wordpress-mobile/WordPress-Android
# Use external browser for all atomic previews to avoid caching issue

This PR addresses one of the sources of the out-of-date previews issue reported in #13361. In particular, it addresses the out-date-preview reported in [this comment](https://github.com/wordpress-mobile/WordPress-Android/issues/13361#issuecomment-1063699069) on that issue.

This happens because we are making unauthenticated calls to load the preview in the webview, and that results in the server returning cached responses and sometimes giving previews that do not reflect the latest changes (internal ref: pbArwn-3XS-p2).

We addressed [a somewhat similar issue dealing with published posts](https://github.com/wordpress-mobile/WordPress-Android/issues/13093) earlier in https://github.com/wordpress-mobile/WordPress-Android/pull/13217 and found that it to be very difficult to authenticate through the webview because it kept redirecting to wp-admin, so instead we adopted the approach of loading the preview in the regular browser instead and requiring the user to authenticate through the browser. That fix was only applied to published posts, however.

In this PR, I am removing the check that limits that behavior to published posts. Instead, all atomic site post previews are loaded in the external browser now. 

Although this is certainly not ideal behavior (requiring the user to authenticate in the browser if they haven't authenticated previously), it seems much better than the current behavior (out-of-date previews), and this is a very quick and easy fix to implement until we find a better solution.


### To Test

#### Verify Previews Are Up-To-Date

1. Go to an Atomic site with a plugin installed (so you can be sure it's fully transformed)
1. Go to the drafts tab of the post list and tap the (+) fab to create a new post
2. Type the title "Test"
3. Tap the "Start writing..." placeholder and type a paragraph with the word "Test"
4. Open the preview (tap the three dots menu --> tap "Preview")
5. Browser (not a WebView) opens.
6. If you have not previously authenticated in your browser, you will be asked to authenticate. Do so.
7. Observe the preview (containing the content "Test") opens
8. Go back to the editor
9. Tap at the end of the paragraph block and type " test" (the paragraph should now include the content: "Test test")
10. Open the preview
11. Observe the preview (containing the content "Test test")
12. Go back to the editor
13. Tap at the end of the paragraph block and type " test" (the paragraph should now include the content: "Test test test")
14. Open the preview
15. Observe the preview is up-to-date (contains the content "Test test test")

#### Verify No Regressions in https://github.com/wordpress-mobile/WordPress-Android/pull/13217

Perform the same steps as above, but with a published or scheduled post.

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)
Went through test steps in https://github.com/wordpress-mobile/WordPress-Android/pull/13217

3. What automated tests I added (or what prevented me from doing so)
Did not add any automated tests. This class is not well-designed for testing (static methods calling static methods) and the functionality here is so closely tied to the Android framework that I don't think the cost/benefit of rewriting this to make it more testable is worth it at this point. Let me know if you disagree though. 

PR submission checklist:
